zerver/lib/upload.py

```diff
@@ -380,15 +380,28 @@ def get_signed_upload_url(path: str) -> str:
 class S3UploadBackend(ZulipUploadBackend):
     def __init__(self) -> None:
         self.session = boto3.Session(settings.S3_KEY, settings.S3_SECRET_KEY)
-
         self.avatar_bucket = get_bucket(settings.S3_AVATAR_BUCKET, self.session)
-        network_location = urllib.parse.urlparse(
-            self.avatar_bucket.meta.client.meta.endpoint_url
-        ).netloc
-        self.avatar_bucket_url = f"https://{self.avatar_bucket.name}.{network_location}"
-
         self.uploads_bucket = get_bucket(settings.S3_AUTH_UPLOADS_BUCKET, self.session)
 
+    def get_public_upload_url(
+        self,
+        key: str,
+    ) -> str:
+        config = Config(signature_version=botocore.UNSIGNED)
```

Comment:
I added a comment documenting this, since it's a bit puzzling how this works, and also deduplicated the copy we had in one of the data export paths of this exact logic.